lib/util/cache/package/cacheable.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { DateTime } from 'luxon';
+import { Cacheable } from './cacheable';
+
+describe('util/cache/package/cacheable', () => {
+  it('constructs default value', () => {
+    const res = Cacheable.empty();
+    expect(res.ttlMinutes).toBe(15);
+  });
+
+  describe('TTL', () => {
+    it('static method for minutes', () => {
+      const res = Cacheable.forMinutes(123);
+      expect(res.ttlMinutes).toBe(123);
+    });
+
+    it('method for minutes', () => {
+      const res = Cacheable.empty();
+      expect(res.forMinutes(42).ttlMinutes).toBe(42);
+    });
+
+    it('setter for minutes', () => {
+      const res = Cacheable.empty();
+      res.ttlMinutes = 42;
+      expect(res.ttlMinutes).toBe(42);
+    });
+
+    it('static method for hours', () => {
+      const res = Cacheable.forHours(3);
+      expect(res.ttlMinutes).toBe(180);
+    });
+
+    it('method for hours', () => {
+      const res = Cacheable.empty();
+      expect(res.forHours(3).ttlMinutes).toBe(180);
+    });
+
+    it('setter for hours', () => {
+      const res = Cacheable.empty();
+      res.ttlHours = 3;
+      expect(res.ttlMinutes).toBe(180);
+    });
+
+    it('static method for days', () => {
+      const res = Cacheable.forDays(2);
+      expect(res.ttlMinutes).toBe(2880);
+    });
+
+    it('method for days', () => {
+      const res = Cacheable.empty();
+      expect(res.forDays(2).ttlMinutes).toBe(2880);
+    });
+
+    it('setter for days', () => {
+      const res = Cacheable.empty();
+      res.ttlDays = 2;
+      expect(res.ttlMinutes).toBe(2880);
+    });
+  });
+
+  describe('public data', () => {
+    it('via static method', () => {
+      const res: Cacheable<number> = Cacheable.asPublic(42);
+      expect(res.value).toBe(42);
+      expect(res.isPublic).toBeTrue();
+      expect(res.isPrivate).toBeFalse();
+    });
+
+    it('via method', () => {
+      const res: Cacheable<number> = Cacheable.empty().asPublic(42);
```

Comment:
```suggestion
      const res = Cacheable.empty().asPublic(42);
```
I think this isn't required

Reply:
Yes, but it ensures the right type

Reply:
Because `Cacheable.empty()` returns `Cacheable<unknown>`, and we want to make sure `asPublic(42)` makes correct type inference/conversion.